packages/ws-client/src/data-source.ts

```diff
@@ -112,8 +141,14 @@ export function createWSDataSource(wsUrl: string): DataSource {
       stores.set(key, newStore)
       return newStore
     },
-    sendEvent: (event) => {
+    sendEvent: async (event) => {
       send({ $: 'evt', event })
+      if (isHandlerEvent(event)) {
+        return new Promise((resolve) => {
+          // tbd: do we want to register a timeout here?
```

Comment:
yes I think we should have some timeout, if it is configurable. Maybe the event definition can have an optional timeout

Reply:
yes, let's allow optional timeout to overwrite on per event basis. What should be the default tho?

Reply:
will do a timeout in a follow-up + some unit tests